dali/imgcodec/decoders/decoder_parallel_impl.h

```diff
@@ -74,13 +97,69 @@ class DLL_PUBLIC BatchParallelDecoderImpl : public ImageDecoderImpl {
     std::vector<DecodeResult> ret(out.size());
     ROI no_roi;
     for (int i = 0; i < in.size(); i++) {
-      tp_->AddWork([&, i](int) {
-        ret[i] = Decode(out[i], in[i], opts, rois.empty() ? no_roi : rois[i]);
+      tp_->AddWork([&, i](int tid) {
+        ret[i] = DecodeImplTask(tid, out[i], in[i], opts, rois.empty() ? no_roi : rois[i]);
+      }, volume(out[i].shape()));
+    }
+    tp_->RunAll();
+    return ret;
+  }
+
+  std::vector<DecodeResult> ParallelDecodeImpl(cudaStream_t stream,
+                                               span<SampleView<GPUBackend>> out,
+                                               cspan<ImageSource *> in,
+                                               DecodeParams opts,
+                                               cspan<ROI> rois) {
+    assert(out.size() == in.size());
+    assert(rois.empty() || rois.size() == in.size());
+    std::vector<DecodeResult> ret(out.size());
+    ROI no_roi;
+    for (int i = 0; i < in.size(); i++) {
+      tp_->AddWork([&, i](int tid) {
+        ret[i] = DecodeImplTask(stream, tid, out[i], in[i], opts, rois.empty() ? no_roi : rois[i]);
```

Comment:
```suggestion
        ret[i] = DecodeImplTask(tid, stream, out[i], in[i], opts, rois.empty() ? no_roi : rois[i]);
```
The order of arguments is wrong.

Reply:
true. Thanks for spotting that